skt/reporter.py

```diff
@@ -748,6 +751,8 @@ def report(self):
             msg.attach(MIMEText(self.getreport()))
 
         for (name, att) in self.attach:
+            if self.dry_run:
```

Comment:
Wouldn't it be better to skip checking the attachments if the dry run option is detected instead of iterating over them and skipping attaching?
```
if not self.dry_run:
    for (name, att) in self.attach:
        .....
```